modin/core/dataframe/pandas/dataframe/dataframe.py

```diff
@@ -2239,6 +2490,48 @@ def concat(self, axis, others, how, sort):
             new_partitions, new_index, new_columns, new_lengths, new_widths, new_dtypes
         )
 
+    def groupby(
```

Comment:
Should we replace `groupby_reduce` below to this `groupby`?

Reply:
I don't think so - `groupby_reduce` requires `by` to be `None` or another Modin DataFrame, while `groupby` is a more general groupby, so I think the two are different!

Reply:
When do we plan to use this `groupby`?